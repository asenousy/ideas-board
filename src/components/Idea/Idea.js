import React from "react";
import PropTypes from "prop-types";
import styles from "./Idea.module.css";

const titlePlaceHolder = "untitled";

export const Idea = props => {
  const { title, description, date } = props;

  return (
    <div className={styles.wrapper}>
      <h3>{title || titlePlaceHolder}</h3>
      <p>{description}</p>
      <div className={styles.date}>
        {new Date(date)
          .toTimeString()
          .split(" ")
          .shift()}
      </div>
    </div>
  );
};

Idea.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  date: PropTypes.number
};

export default React.memo(Idea);
