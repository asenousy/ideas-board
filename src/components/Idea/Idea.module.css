.wrapper {
  margin: auto;
  text-align: center;
  min-width: 200px;
}

.date {
  position: absolute;
  margin: 5px;
  bottom: 0;
  left: 0;
  font-size: 12px;
}
