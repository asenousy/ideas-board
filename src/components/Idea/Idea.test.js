import React from "react";
import { shallow } from "enzyme";
import Idea from "./Idea";

let wrapper;
beforeEach(() => (wrapper = shallow(<Idea />)));

it("renders without crashing", () => {
  wrapper.setProps({
    title: "title",
    descripton: "descrtiption",
    date: 1560981762261
  });
  expect(wrapper).toMatchSnapshot();
});
