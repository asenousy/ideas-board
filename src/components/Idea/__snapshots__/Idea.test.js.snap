// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED
    date={1560981762261}
    descripton="descrtiption"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="text"
        >
          title
        </h3>,
        <p
          className="text"
        />,
        <div
          className="date"
        >
          23:02:42
        </div>,
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
          "className": "text",
        },
        "ref": null,
        "rendered": "title",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "23:02:42",
          "className": "date",
        },
        "ref": null,
        "rendered": "23:02:42",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="text"
          >
            title
          </h3>,
          <p
            className="text"
          />,
          <div
            className="date"
          >
            23:02:42
          </div>,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "text",
          },
          "ref": null,
          "rendered": "title",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "23:02:42",
            "className": "date",
          },
          "ref": null,
          "rendered": "23:02:42",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
